app/actions/websocket/channel.ts

```diff
@@ -28,14 +92,25 @@ export async function handleUserRemovedEvent(serverUrl: string, msg: any) {
         return;
     }
 
-    if (user.id === msg.data.user_id) {
-        localRemoveUserFromChannel(serverUrl, msg.data.channel_id);
+    const userId = msg.data.user_id;
+    const channelId = msg.data.channel_id;
+
+    const models: Model[] = [];
+
+    if (user.isGuest) {
+        const {models: updateVisibleModels} = await updateUsersNoLongerVisible(serverUrl);
```

Comment:
`updateserNoLogongerVisible` updates the database inside the function, so here you need to pass the argument so it is only prepared